File: baekjoon/js/2839.js
const fs = require("fs");
const buf = fs.readFileSync(0, "utf-8").toString().trim();
let n = Number(buf);

let result = -1;

if (n >= 3 && n % 3 === 0) {
  result = n / 3;
}

let count5 = 0;
while (n >= 5) {
  n -= 5;
  count5++;

  if (n % 3 === 0) {
    result = count5 + n / 3;
  }
}

console.log(result);

File: baekjoon/rust/2839.rs
use std::io::stdin;

fn main() {
    let mut buf = String::new();
    stdin().read_line(&mut buf).unwrap();

    let mut n: u16 = buf.trim().parse().unwrap();

    let mut count = 0;

    while (n >= 3) {
        match n % 5 == 0 {
            true => {
                count += 1;
                n -= 5;
            }
            false => {
                count += 1;
                n -= 3;
            }
        }
    }

    let result = if n == 0 { count } else { -1 };

    print!("{}", result);
}
